applications/ecallmgr/src/ecallmgr_call_control.erl

```diff
@@ -522,25 +546,57 @@ force_queue_advance(#state{call_id=CallId
         'false' ->
             %% if the node is down, don't inject the next FS event
             lager:debug("not continuing until the media node becomes avaliable"),
-            State#state{current_app='undefined'};
+            State#state{current_app='undefined', current_cmd_uuid='undefined'};
         {'empty', _} ->
             lager:debug("no call commands remain queued, hibernating"),
-            State#state{current_app='undefined'};
+            State#state{current_app='undefined', current_cmd_uuid='undefined'};
         {{'value', Cmd}, CmdQ1} ->
-            AppName = kz_json:get_value(<<"Application-Name">>, Cmd),
-            _ = case CallUp
-                    orelse is_post_hangup_command(AppName)
-                of
-                    'true' ->
-                        execute_control_request(Cmd, State);
-                    'false' ->
-                        lager:debug("command '~s' is not valid after hangup, skipping", [AppName]),
-                        maybe_send_error_resp(CallId, Cmd),
-                        self() ! {'force_queue_advance', CallId}
-                end,
-            MsgId = kz_json:get_value(<<"Msg-ID">>, Cmd),
-            State#state{command_q=CmdQ1, current_app=AppName, current_cmd=Cmd
-                       ,keep_alive_ref=get_keep_alive_ref(State), msg_id=MsgId}
+            AppName = kz_json:get_ne_binary_value(<<"Application-Name">>, Cmd),
```

Comment:
This block of code looks near identical to [here](https://github.com/2600hz/kazoo/pull/4272/files#diff-fe09d4db26d08554706e348fbaa737a3R698) - can they be put in a common function?